### PR TITLE
Django 2.2 compatibility

### PR DESCRIPTION
**Pull request by petedermott**
Currently cartridge and mezzanine have some issues with Django 2.2 this PR aims to fix that.

In order to pass tests (and generally work) Mezzanine and filebrowser_safe also need modifications. I have test versions of these available at [github.com/fermorltd/mezzanine@2.2-compat](https://github.com/fermorltd/mezzanine/tree/2.2-compat) and [github.com/fermorltd/filebrowser-safe@2.2-compat](https://github.com/fermorltd/filebrowser-safe/tree/2.2-compat) respectively. (I hope to make PR's for these when I am sure they are completely clean of any bespoke code)

All tests pass but this is my first shot at this so feedback / code review is welcome.

EDIT: Might be worth noting that this will probably break if using a namespace is not 'shop' for the app as URLs that were previously reversed without them ([like this](https://github.com/stephenmcd/cartridge/blob/a8c5cdbfcc3306f2a0e9ad94630e394a7c7b1062/cartridge/shop/models.py#L149)) will no longer work.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 language: python
 env:
-  - DJANGO="22"
+  - DJANGO="Django>=2.2,<=3.0"
```
Comment (henri-hulski) [inline .travis.yml]: Shouldn't it be
```yml
    DJANGO="Django>=2.2,<3.0"
```
?

Comment (henri-hulski) [reply .travis.yml]: Now you get Django 3.0.

Comment (petedermott) [reply .travis.yml]: Good shout, brain dead from Friday afternoon!

**Comment by henri-hulski** [comment]
Most of this issues should be fixed in version 1.0.0b1.
If you find unresolved problems please open a new PR.

**Pull request closed without merging**